src/hooks/useFetchPokemons.js

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from 'react'
+import axios from 'axios'
+import { BASE_URL } from '../config/constants'
+
+export const useFetchPokemons = (itemsPerPage, offset) => {
+  const url = `${BASE_URL}?limit=${itemsPerPage}&offset=${offset}`
+
+  const [data, setData] = useState([])
+  const [dataCount, setDataCount] = useState(0)
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState(null)
+
+  useEffect(() => {
+    setLoading(true)
+
+    const fetchPokeDetails = (data) => {
+      let pageData = []
+      data.forEach((item) =>
```

Comment:
That is a lot of calls you are making and the user might not even want all those details. Please make a call only when a user enters pokemon detail page

Reply:
Ok.
So, shall I display pokemon's name on the table only?

Reply:
The response I get when I first fetch is an object with pokemon name and a URL which I can fetch the details.

Reply:
`0: {
          name: "bulbasaur"
          url: "https://pokeapi.co/api/v2/pokemon/1/"
     }
          `

That is the kind of response I get from `https://pokeapi.co/api/v2/pokemon/`. I think it makes sense to call the URL I receive to get more information about the pokemon.  Otherwise, I'll have a table only with the pokemon names. I think you ask me for a table with id, name, type, height and weight.